signer/tests/integration/block_observer.rs

```diff
@@ -331,3 +354,350 @@ async fn link_blocks() {
 
     testing::storage::drop_db(db).await;
 }
+
```

Comment:
Hm, I think I would have preferred that these returned a `Vec<TxOutput>` and `Vec<TxPrevout>` and used `.is_empty()` instead of `.is_none()` etc. below, to encourage re-use, since the only filter is on the type (another future test might expect two of them, for example).